demos/common/python/models/deblurring.py

```diff
@@ -74,15 +75,10 @@ def prepare_outputs(self):
     def preprocess(self, inputs):
         image = inputs
 
-        if image.shape[0] < self.h and image.shape[1] < self.w:
-            # right bottom padding to resize input image to input_layer shape
-            pad_params = {'mode': 'constant',
-                          'constant_values': 0,
-                          'pad_width': ((0, self.h - image.shape[0]), (0, self.w - image.shape[1]), (0, 0))
-                          }
-            resized_image = np.pad(image, **pad_params)
-        else:
-            resized_image = cv2.resize(image, (self.w, self.h))
+        if not (self.h - self.block_size < image.shape[0] <= self.h and self.w - self.block_size < image.shape[1] <= self.w):
+            self.logger.warn("Chosen model size doesn't match image size. The image is resized")
+
+        resized_image = cv2.resize(image, (self.w, self.h))
```

Comment:
The image is resized even if the shape differs a bit. Should the warning be triggered even if a shape is in `block_size` range but doesn't match exactly?

Reply:
There are 2 variants:
1) Currently. If image shape in range `(input_layer - bs; input_layer + bs)` we do micro resize, but it does not affect accuracy (tested by AC). So I am not liable to print warn. For other images (out of range), I can't guarantee the accuracy. So for them, print the warn.
2) Do check `if image.shape != input_layer.shape` for match exactly and warn for all images that require resize.

I am liable to choose first variant.

Reply:
> tested by AC

It doesn't mean that images that would produce different results don't exist (or can't be created :) ).

I'm not familiar with the net, so it's up to you.

If you keep your approach, the warning message should be modified so it wouldn't be misleading because it's very tempting to assume that if the warning isn't reported, the image isn't resized.

Reply:
Ok, I went back to the previous method. The last question is what about VideoWriter. It works with constant shape and if we use folder with images with different shapes, VideoWriter will work only with images which shape is equal to the first image shape.

Reply:
Should the prediction be cropped if the image is padded?

VideoWriter works with arbitrary images if filename is % based pattern. I don't think we should do anything about it.

Reply:
Add crop

Reply:
You can't rely on `Deblurring` state to correspond to current postprocessed image because `Deblurring` could preprocess multiple images before inference fishes, so `self.padded` can't be trusted. `posptorocess()` already has image size so it should deduce on its own if the image was padded.

Reply:
Ok, thank you for noticing.